metals/src/main/scala/scala/meta/internal/metals/debug/ClientConfigurationAdapter.scala

```diff
@@ -35,6 +38,37 @@ private[debug] final case class ClientConfigurationAdapter(
     sourceMapper.mappedLineForClient(path, adaptedLine)
   }
 
+  /**
+   * In the DAP specification, the presentationHint of a StackFrame can be
+   * 'normal', 'label' or 'subtle'. Most DAP implementations use 'subtle' to
+   * indicate that a frame is skipped by the debugger. The problem is that
+   * VSCode does not collapse 'subtle' frames, as other DAP clients do.
+   * Instead it collapses 'deemphasize' frames, even if it is not part of the
+   * spec.
+   *
+   * See https://github.com/microsoft/vscode/issues/206801
+   */
+  def adaptStackTraceResponse(result: JsonObject): JsonObject = {
+    if (clientId.contains("vscode")) {
+      try {
+        // For VSCode only, we hack the json result of the stack trace response
+        // to replace all occurrences of 'subtle' by 'deemphasize'.
+        val frames = result.get("stackFrames").getAsJsonArray()
+        for (i <- 0.until(frames.size)) {
+          val frame = frames.get(i).getAsJsonObject()
+          val presentationHint = Option(frame.get("presentationHint"))
+            .map(_.getAsJsonPrimitive.getAsString)
+          if (presentationHint.contains("subtle")) {
+            frame.add("presentationHint", new JsonPrimitive("deemphasize"))
+          }
+        }
+      } catch {
+        case _: IllegalStateException => ()
```

Comment:
```suggestion
        case NonFatal(t) => scribe.warn("unexpected error when adapting stack trace response", t)
```
Maybe to be on the safe side?